### ISSUE TITLE
Wrong queue name when config.active_job.queue_name_prefix is set.

### ISSUE THREAD

**Issue by aruprakshit** [moove-it/sidekiq-scheduler#203]
When u have this setup

```ruby
config.active_job.queue_name_prefix = "muse"
```

Then the actual queue name is being created as `muse_muse_<queue_name>`. The source of bug is

this [line](https://github.com/moove-it/sidekiq-scheduler/blob/5953fdfa3c7574b9aeb89bed96a232456600de89/lib/sidekiq/scheduler.rb#L240) . From here we pass a prefixed queue name and then Rails adds the prefix again to the queue name from [here](https://github.com/rails/rails/blob/cfb1e4dfd8813d3d5c75a15a750b3c53eebdea65/activejob/lib/active_job/enqueuing.rb#L47)

I have created a [sample app](https://gitlab.com/aruprakshit/sidekiq-scheduler-sample). Look at logs

# Sidekiq log

```sh
2017-07-29T21:13:31.883Z 10842 TID-ov8xv042w INFO: Running in ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
2017-07-29T21:13:31.883Z 10842 TID-ov8xv042w INFO: See LICENSE and the LGPL-3.0 for licensing details.
2017-07-29T21:13:31.883Z 10842 TID-ov8xv042w INFO: Upgrade to Sidekiq Pro for more features and support: http://sidekiq.org
2017-07-29T21:13:31.909Z 10842 TID-ov8xv042w INFO: Loading Schedule
2017-07-29T21:13:31.909Z 10842 TID-ov8xv042w INFO: Scheduling SampleJob {"every"=>"15s", "class"=>"SampleJob", "queue"=>"muse_milestones"}
2017-07-29T21:13:31.910Z 10842 TID-ov8xv042w INFO: Schedules Loaded
2017-07-29T21:13:31.910Z 10842 TID-ov8xv042w INFO: Starting processing, hit Ctrl-C to stop
2017-07-29T21:13:47.096Z 10842 TID-ov8ya3lg8 INFO: queueing SampleJob (SampleJob)
```

# development.log

```sh
[ActiveJob] Enqueued SampleJob (Job ID: 1bf988a5-f147-4ee9-b4f0-ffa2fc605786) to Sidekiq(muse_muse_milestones)
[ActiveJob] Enqueued SampleJob (Job ID: 56d2f84d-565d-47c6-bd41-0887a604a6b1) to Sidekiq(muse_muse_milestones)
[ActiveJob] Enqueued SampleJob (Job ID: b9c9dc9e-bc99-4aae-85ef-db55126a7773) to Sidekiq(muse_muse_milestones)
```

The log comes from the [line](https://github.com/rails/rails/blob/cfb1e4dfd8813d3d5c75a15a750b3c53eebdea65/activejob/lib/active_job/logging.rb#L60).

Do we really need the option `{ queue: config['queue'] }` to pass ?

```ruby
def enqueue_with_active_job(config)
   options = { queue: config['queue'] }

   initialize_active_job(config['class'], config['args']).enqueue(options)
end
```

I can open a PR if you like my idea. I don't see the point to pass it. It is fine when you don't have prefix set. With prefix, it creates wrong queue name.

**Comment by snmgian** [comment]
Hi @aruprakshit 

I can't access to https://gitlab.com/aruprakshit/sidekiq-scheduler-sample, it gives me a 404 error page.

What do you have in your `config/sidekiq.yml`. It seems that the name of your queue is `muse_milestones`. So, when active job adds the `muse` prefix, it turns to `muse_muse_milestones`

```
2017-07-29T21:13:31.909Z 10842 TID-ov8xv042w INFO: Scheduling SampleJob {"every"=>"15s", "class"=>"SampleJob", "queue"=>"muse_milestones"}
```

This line in the logs comes from here: https://github.com/moove-it/sidekiq-scheduler/blob/master/lib/sidekiq/scheduler.rb#L92, it prints out the config.

I also set up a sample project trying to reproduce the issue, found that the job is enqueued into `muse_some_queue`. 

If you provide a repo to a sample project I'll take a look.

**Comment by aruprakshit** [comment]
@snmgian I made the repo _public_. Sorry for not doing it before hand. You can check now.

Also please enable it as I did `config.active_job.queue_name_prefix`. When u have not set this settings, then all is good. When you set the prefix, sidekiq creates the correct queue name and then send it to ActiveJob as I linked above in the original post. Now ActiveJob again adds the prefix the queue, and it gets messy. Please have a look my repo, and let me know if you need more informations from my end.

**Comment by snmgian** [comment]
Still can't access :(

I don't see it under your profile: https://gitlab.com/users/aruprakshit/projects

**Comment by aruprakshit** [comment]
@santiestra Gitlab issue it was. Now I tested in 2 browsers, and it is there. Sorry :(

**Comment by snmgian** [comment]
@aruprakshit great, I'll take a look

**Comment by snmgian** [comment]
Released in v2.1.8

**Comment by aruprakshit** [comment]
@snmgian Thanks.
